File: src/main/java/no/nav/domain/pensjon/regler/beregning2011/SisteAldersberegning2011.java
package no.nav.domain.pensjon.regler.beregning2011;

import no.nav.domain.pensjon.regler.grunnlag.Beholdninger;
import no.nav.domain.pensjon.regler.grunnlag.Persongrunnlag;
import no.nav.domain.pensjon.regler.kode.BeregningMetodeTypeCti;
import no.nav.domain.pensjon.regler.kode.RegelverkTypeCti;
import no.nav.domain.pensjon.regler.trygdetid.Brok;
import no.nav.domain.pensjon.regler.vedtak.VilkarsVedtak;

/**
 * Beregning på nytt regelverk
 * Denne benyttes av 2011, 2016 og 2025
 * For 2016 vil alle feltene være aktuelle
 * 
 * @author PREG
 * @author Stian Kroknes (Accenture) - PENPORT-250
 * @author Stian Kroknes (Accenture) Rullet tilbake PENPORT-250
 */
public class SisteAldersberegning2011 extends SisteBeregning {


    private RegelverkTypeCti regelverkType; // Dene benyttes for å avgjøre om det er på 2011, 2016 eller 2025.
    private Basispensjon basispensjon; // 2011
    private Basispensjon restpensjon; // 2011
    private PensjonUnderUtbetaling pensjonUnderUtbetaling;
    private PensjonUnderUtbetaling pensjonUnderUtbetaling2025AltKonv; // PensjonUnderUtbetaling for tapende konvensjon i siste aldersberegning.
    private Beholdninger beholdninger; // 2025
    private Beholdninger beholdningerAltKonv; // Beholdninger for tapende konvensjon i siste aldersberegning.
    private Persongrunnlag eps;
    private boolean epsMottarPensjon;
    private VilkarsVedtak vilkarsvedtakEktefelletillegg;
    private Persongrunnlag avdodesPersongrunnlag;
    private boolean gjenlevenderettAnvendt;
    private VilkarsVedtak anvendtGjenlevenderettVedtak;
    private BeregningMetodeTypeCti beregningsMetode;
    private int tt_anv_kap_20;
    private Brok prorataBrok_kap_20;
    private int tt_anv_kap_20AltKonv;
    private Brok prorataBrok_kap_20AltKonv;

}

File: src/main/kotlin/no/nav/pensjon/regler/domain/beregning2011/SisteAldersberegning2011.kt
package no.nav.pensjon.regler.domain.beregning2011

import com.fasterxml.jackson.annotation.JsonSubTypes
import com.fasterxml.jackson.annotation.JsonTypeInfo
import no.nav.pensjon.regler.domain.enum.BeregningsmetodeEnum
import no.nav.pensjon.regler.domain.enum.RegelverkTypeEnum
import no.nav.pensjon.regler.domain.grunnlag.Beholdninger
import no.nav.pensjon.regler.domain.grunnlag.Persongrunnlag
import no.nav.pensjon.regler.domain.trygdetid.Brok
import no.nav.pensjon.regler.domain.vedtak.VilkarsVedtak

/**
 * Beregning på nytt regelverk
 * Denne benyttes av 2011, 2016 og 2025
 * For 2016 vil alle feltene være aktuelle
 */
@JsonSubTypes(
    JsonSubTypes.Type(value = SisteAldersberegning2016::class)
)
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
open class SisteAldersberegning2011 : SisteBeregning() {
    var regelverkTypeEnum: RegelverkTypeEnum? = null // Denne benyttes for å avgjøre om det er på 2011, 2016 eller 2025.
    var basispensjon: Basispensjon? = null // 2011
    var restpensjon: Basispensjon? = null // 2011
    var pensjonUnderUtbetaling: PensjonUnderUtbetaling? = null
    var pensjonUnderUtbetaling2025AltKonv: PensjonUnderUtbetaling? =
        null // PensjonUnderUtbetaling for tapende konvensjon i siste aldersberegning.
    var beholdninger: Beholdninger? = null // 2025
    var beholdningerAltKonv: Beholdninger? = null // Beholdninger for tapende konvensjon i siste aldersberegning.
    var eps: Persongrunnlag? = null
    var epsMottarPensjon = false
    var vilkarsvedtakEktefelletillegg: VilkarsVedtak? = null
    var avdodesPersongrunnlag: Persongrunnlag? = null
    var gjenlevenderettAnvendt = false
    var anvendtGjenlevenderettVedtak: VilkarsVedtak? = null
    var beregningsMetodeEnum: BeregningsmetodeEnum? = null
    var tt_anv_kap_20 = 0
    var prorataBrok_kap_20: Brok? = null
    var tt_anv_kap_20AltKonv = 0
    var prorataBrok_kap_20AltKonv: Brok? = null

}